src/utils/url-test.js

```diff
@@ -258,3 +260,77 @@ describe('addQueryParam', () => {
     expect(addQueryParam('file?d=e#f', 'a', 'b')).to.equal('file?d=e&a=b#f');
   });
 });
+
+describe('serializeProtoMessageForUrl', () => {
```

Comment:
Could we simplify both of these tests to just verify `JSON.stringify(inputArray.slice(1))` matches the result of `serializeProtoMessageForUrl`?

Reply:
There are sub objects and we need to shift each of the sub arrays EXCEPT for a single sub array that is actually a list of strings and not a sub object.

Reply:
Looking at this again, I'm testing an object without sub objects and perhaps I should also test one with sub objects but your suggestion would probably work for these simpler non complicated messages.  

What I'm doing validates it can be deserialized though and that everything can get fed in, serialized, fed out, deserialized and that all the possible outputs are expected.

I do think you could argue some of this is already tested in the api_messages-test.js file.

We may want to have a conversation about this (and the sendBeacon thing).